inference-engine/src/vpu/common/src/ngraph/transformations/extract_dynamic_batch/extract_dynamic_batch.cpp

```diff
@@ -0,0 +1,487 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/opsets/opset5.hpp"
+
+#include "vpu/utils/optional.hpp"
+#include "vpu/ngraph/utilities.hpp"
+
+#include "vpu/ngraph/transformations/extract_dynamic_batch/extract_dynamic_batch.hpp"
+#include "vpu/ngraph/transformations/extract_dynamic_batch/slice_mat_mul.hpp"
+#include "vpu/ngraph/transformations/extract_dynamic_batch/slice_convolution.hpp"
+#include "vpu/ngraph/transformations/extract_dynamic_batch/slice_binary_eltwise.hpp"
+#include "vpu/ngraph/transformations/extract_dynamic_batch/slice_unary_eltwise.hpp"
+
+#include <queue>
+
+namespace vpu {
+
+NGRAPH_RTTI_DEFINITION(vpu::ExtractBatch, "ExtractBatch", 0);
+
+ExtractBatch::ExtractBatch(std::unordered_set<ngraph::Node::type_info_t> targets) : targets(std::move(targets)) {}
+
+namespace {
+
+class Slicers {
+public:
+    static bool isSupported(const ngraph::Node& node) {
+        return getSlicers().count(node.get_type_info());
+    }
+
+    static SliceConfiguration slice(const ngraph::Node& node) {
+        const auto& slicers = getSlicers();
+        const auto& type = node.get_type_info();
+        return slicers.count(type) ? slicers.at(type)(node) : SliceConfiguration{};
+    }
+
+private:
+    using Functor = std::function<SliceConfiguration(const ngraph::Node&)>;
+    static const std::unordered_map<ngraph::DiscreteTypeInfo, Functor>& getSlicers() {
+        static const std::unordered_map<ngraph::DiscreteTypeInfo, Functor>& slicers = {
+            {ngraph::opset5::MatMul::type_info,                  sliceMatMul},
+            {ngraph::opset5::Convolution::type_info,             sliceConvolution},
+            {ngraph::opset5::GroupConvolution::type_info,        sliceConvolution},
+            {ngraph::opset5::ConvolutionBackpropData::type_info, sliceConvolution},
```

Comment:
does `sliceConvolution` suit `ConvolutionBackpropData` ?

Reply:
As I see it from specification - yes. All convolution-like layers have similar specification regarding batch

Reply:
ok